libs/components/forms/src/lib/modules/checkbox/checkbox.modern.component.scss

```diff
@@ -6,10 +6,17 @@
   --sky-override-checkbox-checked-color: var(--sky-color-icon-action);
   --sky-override-checkbox-checked-color-disabled: #{$sky-text-color-default};
   --sky-override-checkbox-indeterminate-font-size: 14px;
+  --sky-override-checkbox-label-margin-right: var(--sky-margin-inline-xs);
 }
 
 @include mixins.sky-component-host('modern') {
   display: inline-block;
+
+  .sky-switch:not(:has(~ .sky-control-help-container)):not(
```

Comment:
I think this could potentially be simplified, maybe
```
.sky-switch:not(:has(~ .sky-control-help-container, .sky-switch-control-icon)):has(sky-checkbox-label-text-label)
```

Reply:
Unfortunately - it can't really. This is an attempt to maintain current behavior that was done via a margin class. That behavior was based on label text being there (`:has(sky-checkbox-label-text-label`) and an icon and help inline not being given `:not(:has(~ .sky-control-help-container, .sky-switch-control-icon))`. 

I wish it was cleaner - but I don't think it is possible as each piece here is looking for a specific portion of the logic.

Reply:
For some reason the `sky-switch` element is on the `label` instead of `sky-switch-label` on checkbox. And then the help and icon are within the label element - so this gets more complicated than it was in radio.